include/sparrow/layout.hpp

```diff
@@ -94,11 +94,14 @@ namespace sparrow
     public:
 
         using self_type = primitive_layout<T>;
-        using value_type = T;
-        using reference = T&;
-        using const_reference = const T&;
-        using pointer = T*;
-        using const_pointer = const T*;
+        using inner_value_type = T;
+        using value_type = std::optional<inner_value_type>;
+        using inner_reference = inner_value_type&;
+        using inner_const_reference = const inner_reference;
+        using reference = reference_proxy<self_type>;
+        using const_reference = const_reference_proxy<self_type>;
+        using pointer = inner_value_type*;
+        using const_pointer = const inner_value_type*;
```

Comment:
These are the definitions for `inner_pointer` and `inner_const_pointer`. I don't think we need the pointer types for now.

Reply:
these types shold be rename `inner_pointer` and `inner_const_pointer`